cmd/spire-agent/cli/run/run.go

```diff
@@ -229,6 +234,65 @@ func mergeInput(fileInput *Config, cliInput *agentConfig) (*Config, error) {
 	return c, nil
 }
 
+func downloadTrustBundle(ac *agent.Config, c *Config) ([]*x509.Certificate, error) {
+	u, err := url.Parse(c.Agent.TrustBundleURL)
+	if err != nil {
+		return nil, fmt.Errorf("could not parse trust bundle URL: %s", u)
+	}
+
+	if strings.ToLower(u.Scheme) != "https" {
+		return nil, errors.New("the trust bundle URL must start with https://")
+	}
+
+	ac.Log.Infof("Fetching trust bundle from remote URL %s", c.Agent.TrustBundleURL)
+	resp, err := http.Get(c.Agent.TrustBundleURL)
+	if err != nil {
+		return nil, fmt.Errorf("unable to fetch trust bundle URL %s: %s", c.Agent.TrustBundleURL, err.Error())
+	}
+
+	defer resp.Body.Close()
+	pemBytes, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("unable to read from trust bundle URL %s: %s", c.Agent.TrustBundleURL, err.Error())
+	}
+
+	bundle, err := pemutil.ParseCertificates(pemBytes)
+	if err != nil {
+		return nil, err
+	}
+
+	return bundle, nil
+}
+
+func setupTrustBundle(ac *agent.Config, c *Config) error {
+	// If trust_bundle_url is set, download the trust bundle using HTTP and parse it from memory
+	// If trust_bundle_path is set, parse the trust bundle file on disk
+	// Both cannot be set
+	// The trust bundle URL must start with HTTPS
+
+	ac.InsecureBootstrap = c.Agent.InsecureBootstrap
+
+	if c.Agent.TrustBundleURL != "" && c.Agent.TrustBundlePath != "" {
```

Comment:
this validation may go into validateConfig(), where we put all validations for our config, what do you thin about that?